### PR TITLE
Update travis to test python 3.6 and 3.7

### PR DESCRIPTION
**Pull request by jchodera**
Yank, omnia, and conda-forge no longer support python 3.5.

This PR updates travis to test Python 3.6 and 3.7, pushing docs on the 3.7 build.

The Python 3.7 build is now correctly tested with the nightly builds of openmm---that had been broken previously because the wrong channel was specified.

### CODE REVIEW

**Comment by jchodera** [comment]
Looks like Python 3.7 is having some trouble:
```
Packaging yank-dev
INFO:conda_build.build:Packaging yank-dev
Packaging yank-dev-0.0.0-py37_0
INFO:conda_build.build:Packaging yank-dev-0.0.0-py37_0
compiling .pyc files...
number of files: 157
patchelf: file: /home/travis/miniconda/conda-bld/yank-dev_1548611818923/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/yank-0.23.8-py3.7-linux-x86_64.egg/yank/multistate/mixing/_mix_replicas.cpython-37m-x86_64-linux-gnu.so
    setting rpath to: $ORIGIN/../../../../../..
WARNING (yank-dev,lib/python3.7/site-packages/yank-0.23.8-py3.7-linux-x86_64.egg/yank/multistate/mixing/_mix_replicas.cpython-37m-x86_64-linux-gnu.so): Needed DSO / not found in any CDT/compiler package, nor the whitelist?!
WARNING (yank-dev,lib/python3.7/site-packages/yank-0.23.8-py3.7-linux-x86_64.egg/yank/multistate/mixing/_mix_replicas.cpython-37m-x86_64-linux-gnu.so): Needed DSO / not found in any CDT/compiler package, nor the whitelist?!
   INFO (yank-dev): plugin library package conda-forge::cython-0.29.3-py37hf484d3e_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
Fixing permissions
Detected hard-coded path in binary file lib/python3.7/site-packages/yank-0.23.8-py3.7-linux-x86_64.egg/yank/multistate/mixing/_mix_replicas.cpython-37m-x86_64-linux-gnu.so
Detected hard-coded path in text file bin/yank
Compressing to /tmp/tmp9RioKx/yank-dev-0.0.0-py37_0.tar.bz2
Importing conda-verify failed.  Please be sure to test your packages.  conda install conda-verify to make this message go away.
WARNING:conda_build.build:Importing conda-verify failed.  Please be sure to test your packages.  conda install conda-verify to make this message go away.
TEST START: /home/travis/miniconda/conda-bld/linux-64/yank-dev-0.0.0-py37_0.tar.bz2
Adding in variants from /tmp/tmpERoMhu/info/recipe/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /tmp/tmpERoMhu/info/recipe/conda_build_config.yaml
Renaming work directory,  /home/travis/miniconda/conda-bld/yank-dev_1548611818923/work  to  /home/travis/miniconda/conda-bld/yank-dev_1548611818923/work_moved_yank-dev-0.0.0-py37_0_linux-64
Collecting package metadata: ...working... done
Solving environment: ...working... failed
failed to get install actions, retrying.  exception was: Unsatisfiable dependencies for platform linux-64: set(['yank-dev==0.0.0=py37_0'])
WARNING:conda_build.build:failed to get install actions, retrying.  exception was: Unsatisfiable dependencies for platform linux-64: set(['yank-dev==0.0.0=py37_0'])
Tests failed for yank-dev-0.0.0-py37_0.tar.bz2 - moving package to /home/travis/miniconda/conda-bld/broken
WARNING:conda_build.build:Tests failed for yank-dev-0.0.0-py37_0.tar.bz2 - moving package to /home/travis/miniconda/conda-bld/broken
TESTS FAILED: yank-dev-0.0.0-py37_0.tar.bz2
```
I'm not quite sure what's going wrong here.

**Comment by jchodera** [comment]
The Python 3.7 error is very odd. It must be that `conda install` has some weird local behavior that doesn't match Python 3.6:
```
# Automatic uploading is disabled
# If you want to upload package(s) to anaconda.org later, type:
anaconda upload /home/travis/miniconda/envs/test/conda-bld/linux-64/yank-dev-0.0.0-py37_0.tar.bz2
# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes
anaconda_upload is not set.  Not uploading wheels: []
####################################################################################
Resource usage summary:
Total time: 0:01:52.2
CPU usage: sys=0:00:00.3, user=0:00:03.7
Maximum memory usage observed: 151.4M
Total disk usage observed (not including envs): 56.3K
The command "conda build devtools/conda-recipe" exited with 0.
29.01s$ conda install -c $HOME/miniconda/envs/test/conda-bld/ --quiet --yes --use-local ${PACKAGENAME}-dev
Collecting package metadata: ...working... done
Solving environment: ...working... failed
UnsatisfiableError: The following specifications were found to be in conflict:
  - yank-dev
Use "conda search <package> --info" to see the dependencies for each package.
```
We may just need to bite the bullet and go for the new cookiecutter (#1132)

**Comment by jchodera** [comment]
Now that I've updated NetCDF, I'm running into [the masked arrays issue here](https://github.com/choderalab/openmmtools/pull/398#issuecomment-458981473).

**Comment by jchodera** [comment]
@andrrizzi : Any idea what these errors on the py37 build might be due to?
https://travis-ci.org/choderalab/yank/jobs/488278695#L2343-L2429

File: .travis.yml [hunk 1]
```diff
@@ -31,25 +31,23 @@ install:
...
-  - cd devtools && nosetests $PACKAGENAME --with-coverage --nocapture --verbosity=2 --with-timer -a '!slow' && cd .. && if [[ $CONDA_PY = 3.5 ]]; then devtools/travis-ci/build_docs.sh; fi
+  - cd devtools && nosetests $PACKAGENAME --with-coverage --nocapture --verbosity=2 --with-timer -a '!slow' && cd .. && if [[ $CONDA_PY = 3.7 ]]; then devtools/travis-ci/build_docs.sh; fi
 
 env:
   matrix:
-    - python=3.5 CONDA_PY=3.5
-    - python=3.5 CONDA_PY=35 DEVOMNIA=true
     - python=3.6 CONDA_PY=3.6
+    - python=3.7 CONDA_PY=3.7
+    - python=3.7 CONDA_PY=3.7 DEVOMNIA=true
```
Comment (andrrizzi) [inline .travis.yml]: Whoops! In the end, I didn't post the comment with the review.

I think the `CONDA_PY` here must be set to the notation `3.7` to make this command work: https://github.com/choderalab/yank/blob/master/devtools/travis-ci/build_docs.sh#L7

**Comment by andrrizzi** [comment]
Hmm. I'm not sure why all these sphinx packages are pinned [here](https://github.com/choderalab/yank/blob/master/devtools/travis-ci/build_docs.sh#L9-L12) and why we install sphinx both through conda and through pip, but I'll try to unpin everything. sphinx 1.5.6 is not built for py37.

**Comment by andrrizzi** [comment]
Actually, it doesn't look like sphinx_rtd_theme is built for py37, I'm going to build the docs for 3.6.

**Comment by andrrizzi** [comment]
Uff, now this happens when installing the docs requirements: https://travis-ci.org/choderalab/yank/jobs/490530299#L5387-L5389 .

Not sure if the syntax will work, but I'll try to install both local yank and conda requiriments at the same time.

**Comment by andrrizzi** [comment]
Finally! Tests are passing!

**Comment by jchodera** [comment]
Hooray! Thank you!

**Pull request merged**